src/Lecture3.hs

```diff
@@ -89,8 +95,23 @@ weekday to the second.
 >>> daysTo Friday Wednesday
 5
 -}
-daysTo = error "TODO"
 
+-- Using this ["trick"](https://t.me/c/1573945457/940) with `forall`
+-- or we could just use Weekday in types as we already have "days" in
+-- function name or even put `7` in place for len in 
+-- otherwise = len + (from2 - from1) :)
```

Comment:
I would specify the type of this function to work explicitly with `Weekday`. Indeed, it enabled some optimizations like being able to use the constant `7` directly. Also, its name `daysTo` is not too generic to work with any enumeration type (unlike `next`).

But sure, it's possible to implement a function with the existing type signatures and call it, let's say, `elementsTo` or something like that 🙂